src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/RoleAssignmentIntegrationTest.java

```diff
@@ -48,5 +74,142 @@ public void shoudGetRecordsFromRoleAssignmentTable() throws Exception {
         assertEquals(
             "Role assignment actor Id", "123e4567-e89b-42d3-a456-556642445613", actorId);
     }
-}
 
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts =
+        {"classpath:sql/insert_role_assignment.sql",
+            "classpath:sql/insert_actor_cache_control.sql"})
+    public void shouldGetRoleAssignmentsBasedOnActorId() throws Exception {
+        assertRoleAssignmentRecordSize();
+        final String url = "/am/role-assignments/actors/123e4567-e89b-42d3-a456-556642445612";
+
+        final MvcResult result = mockMvc.perform(get(url)
+                                                     .contentType(JSON_CONTENT_TYPE)
+                                                     .headers(getHttpHeaders()))
+            .andExpect(status().is(200))
+            .andReturn();
+
+        RoleAssignmentResource response = mapper.readValue(
+            result.getResponse().getContentAsString(),
```

Comment:
It may be possibility that result.getResponse() may be empty or null.So,We should have some check.

Reply:
Added additional check..